### PR TITLE
[TTAHUB-311] Add admin page (& associated policies) for managing feature flags

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
Implement a basic "feature flag" system, including giving the admin users the ability to edit which users have feature flag capabilities.

- Add a column to Users (like roles) for the flags
- Use policies/Users to check access
- Admins still have blanket access
- On the Admin section, features can be removed from the user administration screen, users can be filtered by feature, and I've also added a low-functionality page to simply list active feature flags.  
- Create a component ```<FeatureFlag />``` that will conditionally hide it's children something based on user/selected flag.

There may be the ability to use cloud.gov environmental variables to activate/deactivate feature flags, however in my initial testing I failed to inject them into the front end build step. More research may yield a working method.

## How to test
Running the migrations adds a single feature flag, 'grantee_record_page' to the db. You should be able to add and remove it from users locally. The CI tests verify that the policies and components work as expected.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-311

### Every PR

- [x] Meets issue criteria
- [ ] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Review by jasalisbury — COMMENTED** [review]
I think this looks good so far! I like the ability to filter users to those that have specific feature flags. Some thoughts

- I think it would be nice to have a frontend component that handles looking at the feature flag and conditionally rendering children. So when we are ready to stop using the feature flag someone just removes the "FeatureFlag" component. Something like...
```
<FeatureFlag flagName="regional_dashboard">
  <RegionalDashboard />
</FeatureFlag>
```
- In the same spirit I think the backend could use an explicit "feature flag" policy to guard feature flagged API endpoints. So while featured flagged an endpoint may have more than one policy
- I'd play around with migrations and the process to remove feature flags from the database enumeration type, I remember running into issues removing/updating enumerations in the past.

**Comment by thewatermethod** [comment]
>
> I think this looks good so far! I like the ability to filter users to those that have specific feature flags. Some thoughts
> 
>   I think it would be nice to have a frontend component that handles looking at the feature flag and conditionally rendering children. So when we are ready to stop using the feature flag someone just removes the "FeatureFlag" component. Something like...
> 
> 
> ```
> <FeatureFlag flagName="regional_dashboard">
>   <RegionalDashboard />
> </FeatureFlag>
> ```

I think that ```<FeatureFlag />``` idea is really cool... I'll work on adding a demo of that to this PR.

>
>    In the same spirit I think the backend could use an explicit "feature flag" policy to guard feature flagged API endpoints. So while featured flagged an endpoint may have more than one policy
> 

I'm actually not sure what you mean by this. Something in addition to the ActivityReports.canSeeBehindFeatureFlags(flag) implementation?

>    I'd play around with migrations and the process to remove feature flags from the database enumeration type, I remember running into issues removing/updating enumerations in the past.

Great advice, I will do that. Also leaving this here for my own notes JIC 
- https://github.com/abelosorio/sequelize_replace_enum_postgres

**Comment by jasalisbury** [comment]
> > In the same spirit I think the backend could use an explicit "feature flag" policy to guard feature flagged API endpoints. So while featured flagged an endpoint may have more than one policy
> 
> I'm actually not sure what you mean by this. Something in addition to the ActivityReports.canSeeBehindFeatureFlags(flag) implementation?

I'm proposing adding a "FeatureFlag" policy class with a method _like_ `FeatureFlag.hasFlag("regional_dashboard")` (name could be better). So after we are done with the feature flag we remove the instantiation and usage of the "FeatureFlag" policy class and the actual implementation of the feature remains unchanged. I like it being a separate policy class because we won't always want to add feature flags around just activity reports.

**Comment by thewatermethod** [comment]
> I like it being a separate policy class because we won't always want to add feature flags around just activity reports.

That makes good sense

**Comment by thewatermethod** [comment]
What do you think of adding canSeeBehindFeatureFlag to the user policy, since it is bound up with the users table? We can change the name of that function if its too cute, also not opposed to putting it into it's own policy if we think it merits it

**Comment by jasalisbury** [comment]
> What do you think of adding canSeeBehindFeatureFlag to the user policy, since it is bound up with the users table? We can change the name of that function if its too cute, also not opposed to putting it into it's own policy if we think it merits it

Yeah I think that works!

File: frontend/src/pages/Admin/Flags.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        setFeatures(featuresFromApi);
+      } catch (e) {
+        setError('Unable to fetch features');
+      }
+    }
+
+    if (!features) {
+      fetchFeatures();
+    }
+  }, [features]);
```
Comment (jasalisbury) [inline frontend/src/pages/Admin/Flags.js:25]: If you give this `useEffect` an empty dependency array you can drop the check on features existing

File: frontend/src/pages/Admin/users.js [hunk 2]
```diff
@@ -61,6 +80,21 @@ function Admin(props) {
...
+    async function fetchFeatures() {
+      try {
+        const featureOptions = await getFeatures();
+        setFeatures(featureOptions.map((option) => ({ label: option, value: option })));
+      } catch (e) {
+        updateError('Unable to fetch features');
+      }
+    }
+
+    if (!features) {
```
Comment (jasalisbury) [inline frontend/src/pages/Admin/users.js:93]: Can drop the features check, similar as above

**Review by jasalisbury — APPROVED** [review]
LGTM!

**Pull request merged**